internal/stage_common.go

```diff
@@ -1,6 +1,31 @@
 package internal
 
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+)
+
 const URL = "http://127.0.0.1:4221/"
-const TCP_DEST = "localhost:4221"
+const TCP_DEST = "127.0.0.1:4221"
 const DATA_DIR = "/tmp/data/codecrafters.io/http-server-tester/"
 const FILENAME_SIZE = 40
+
+func decodeGZIP(encodedString []byte) (string, error) {
+	reader := bytes.NewReader([]byte(encodedString))
+	gzipReader, err := gzip.NewReader(reader)
+	if err != nil {
+		return "", fmt.Errorf("Failed to create gzip reader: %v", err)
```

Comment:
@ryan-gang this error message doesn't mean much to a user. What's a "Gzip reader"? 

We should just say `Failed to decompress data` here since it looks like the error could trigger if the Zlib header isn't valid. If users hit this a lot, we might need to wrap these errors and make them more friendly (I imagine the default Go err strings aren't going to be very helpful)